substrate/fd

```diff
@@ -57,7 +55,13 @@ namespace substrate
 #endif
 		fd_t(fd_t &&fd_) noexcept : fd_t{} { swap(fd_); }
 		~fd_t() noexcept { if (fd != -1) close(fd); }
-		void operator =(fd_t &&fd_) noexcept { swap(fd_); }
+
+		fd_t &operator =(fd_t &&fd_) noexcept
+		{
+			swap(fd_);
```

Comment:
This complete type is both MoveConstructible and MoveAssignable (by virtue of all its member being integral literals), so you should be able to just turn all of these into ` = default`s, and rely on `std::swap` where necessary.

See https://en.cppreference.com/w/cpp/algorithm/swap, "Type requirements" section.

Reply:
Removing this because it seems to not be the case in MSVC. I've asked in includec++ for more information.